cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableDataLogRepairCommand.java

```diff
@@ -0,0 +1,914 @@
+        long storageLength = getLongUserInput("Introduce the storage length of the Segment: ");
+        boolean sealed = getBooleanUserInput("Is the Segment sealed? [true/false]: ");
+        boolean sealedInStorage = getBooleanUserInput("Is the Segment sealed in storage? [true/false]: ");
+        boolean deleted = getBooleanUserInput("Is the Segment deleted? [true/false]: ");
+        boolean deletedInStorage = getBooleanUserInput("Is the Segment deleted in storage? [true/false]: ");
+        output("You are about to start adding Attributes to the SegmentProperties instance.");
+        boolean finishInputCommands = !confirmContinue();
+        Map<AttributeId, Long> attributes = new HashMap<>();
+        while (!finishInputCommands) {
+            output("Creating an AttributeUpdateCollection for this operation.");
+            try {
+                AttributeId attributeId = AttributeId.fromUUID(UUID.fromString(getStringUserInput("Introduce UUID for this Attribute: ")));
+                long value = getLongUserInput("Introduce the Value for this Attribute:");
+                attributes.put(attributeId, value);
+            } catch (NumberFormatException ex) {
+                outputError("Wrong input argument.");
+                ex.printStackTrace();
+            } catch (Exception ex) {
+                outputError("Some problem has happened.");
+                ex.printStackTrace();
+            }
+            output("You can continue adding AttributeUpdates to the AttributeUpdateCollection.");
+            finishInputCommands = !confirmContinue();
+        }
+        long lastModified = getLongUserInput("Introduce last modified timestamp for the Segment (milliseconds): ");
+        return StreamSegmentInformation.builder().name(segmentName).startOffset(offset).length(length).storageLength(storageLength)
+                .sealed(sealed).deleted(deleted).sealedInStorage(sealedInStorage).deletedInStorage(deletedInStorage)
+                .attributes(attributes).lastModified(new ImmutableDate(lastModified)).build();
+    }
+
+    /**
+     * Method to create a {@link AttributeUpdateCollection} object to fill the {@link Operation}s that require it.
+     *
+     * @return New {@link AttributeUpdateCollection} object with user-defined content.
+     */
+    @VisibleForTesting
+    AttributeUpdateCollection createAttributeUpdateCollection() {
+        AttributeUpdateCollection attributeUpdates = new AttributeUpdateCollection();
+        output("You are about to start adding AttributeUpdates to the AttributeUpdateCollection.");
+        boolean finishInputCommands = !confirmContinue();
+        while (!finishInputCommands) {
+            output("Creating an AttributeUpdateCollection for this operation.");
+            try {
+                AttributeId attributeId = AttributeId.fromUUID(UUID.fromString(getStringUserInput("Introduce UUID for this AttributeUpdate: ")));
```

Comment:
Not all attributes are UUIDs. The `FixedKeyLengthTableSegment` (find the class that starts with that prefix) uses different Attribute ID lengths. 

Maybe we can improve on this later, but since those segments are also part of the container, we should figure out a way to fix the recovery process if it's stuck on one of them.

Reply:
Right, if you believe that this is an important limitation, I can think on how to do this. If not, we can think about this on a further iteration of the tool.